src/lib/agent/tools/base/Tool.ts

```
@@ -0,0 +1,120 @@
+    }
+  }
+
+  // Validate parameters against the tool's schema
+  protected validateParams(params: unknown): TParams {
+    return this.definition.parameters.parse(params) as TParams
+  }
+
+  // Request approval for tool execution (can be overridden)
+  protected async requestApproval(
+    params: TParams, 
+    context: AgentContext
```

ESLint

[error] 'context' is defined but never used. Allowed unused args must match /^_/u.

(@typescript-eslint/no-unused-vars)